### ISSUE TITLE
ceylon command line plugin failing at Bootstrap

### ISSUE THREAD

**Issue by loicrouchon** [ceylon/ceylon-compiler#2137]
Install a ceylon command line plugin (like `ceylon build` or `ceylon format`)

Launch the plugin command `ceylon build whatever`

It's failing with the following exception:
```
Fatal: Ceylon command could not be executed
Exception in thread "main" java.lang.NullPointerException
	at com.redhat.ceylon.launcher.Bootstrap.run(Bootstrap.java:50)
	at com.redhat.ceylon.launcher.Bootstrap.main(Bootstrap.java:30)
ceylon build: Script /home/loic/.ceylon/bin/ceylon.build.runner/ceylon-build
returned error exit code 1
```

This because `mainMethod` is null in  `Bootstrap` class.

adding echos in the ceylon-build script to know what shell commands were sent returns this:
```
. /home/loic/dev/ceylon/latest/ceylon-dist/dist/bin/ceylon-sh-setup
/home/loic/dev/ceylon/latest/ceylon-dist/dist/bin/ceylon run ceylon.build.runner/1.1.1 whatever
```
Executing those two lines succeed to launch ceylon build which is strange.

To summarize, `ceylon build` fail with a NPE in `Boostrap` but
```
. /home/loic/dev/ceylon/latest/ceylon-dist/dist/bin/ceylon-sh-setup
/home/loic/dev/ceylon/latest/ceylon-dist/dist/bin/ceylon run ceylon.build.runner/1.1.1 whatever
```
is working fine

**Comment by FroMage** [comment]
Confirmed

**Comment by FroMage** [comment]
So this is due to recent changes by @quintesse in `Bootstrap` which uses `CEYLON_VERSION` from the environment to determine the ceylon version to run, and in our scripts, we set `CEYLON_VERSION` to `1.1.1 (Pif Gadget)`, so not the format `Bootstrap` expects. I suppose we should fix the script environment if we intend to support this env var. Any other env var I should know about, @quintesse? Are these documented anywhere?

**Comment by FroMage** [comment]
Fixed, but I'd like @quintesse to review my fix in `Boostrap` because although I'm pretty sure it was wrong to catch the exception (and go on and not exit with an error exit code), I'm not sure if that code is invoked from other places.

**Comment by FroMage** [comment]
Assigning to @quintesse for check, but @loicrouchon you can already start testing: it should work.

**Comment by quintesse** [comment]
> Any other env var I should know about, @quintesse? Are these documented anywhere?

The only ones I know of are defined in `Constants` as `ENV_XXXX` (so there's only `CEYLON_HOME` and `CEYON_VERSION` for now)

**Comment by FroMage** [comment]
OK thanks. So we can close this then.

**Comment by quintesse** [comment]
Changed the code to deal better with exit codes and although I would have preferred perhaps to keep hiding any stacktraces that early in the startup I don't care too much.

**Comment by loicrouchon** [comment]
Working fine, thanks
